diff --git forkSrcPrefix/include/wx/osx/taskbarosx.h forkDstPrefix/include/wx/osx/taskbarosx.h
index bce68c55515c27c4e0690fa2fb31d40ba7ae8f7f..86e04f6972e8d6a642cf3155235cfbde18660792 100644
--- forkSrcPrefix/include/wx/osx/taskbarosx.h
+++ forkDstPrefix/include/wx/osx/taskbarosx.h
@@ -22,6 +22,8 @@ public:

     // returns true if the taskbaricon is in the global menubar
 #if wxOSX_USE_COCOA
+    void SetTitle(const wxString& title);
+    void SetIcon(const wxString& nsImageName);
     bool OSXIsStatusItem();
 #else
     bool OSXIsStatusItem() { return false; }
diff --git forkSrcPrefix/src/osx/cocoa/taskbar.mm forkDstPrefix/src/osx/cocoa/taskbar.mm
index 7941a783c5c4b398587a6a8878583ad25f9ea0dd..9adb6c93f8a140dc7bc43e88d7612e7cebadeb9c 100644
--- forkSrcPrefix/src/osx/cocoa/taskbar.mm
+++ forkDstPrefix/src/osx/cocoa/taskbar.mm
@@ -61,6 +61,8 @@ public:

     virtual bool IsStatusItem() const { return false; }

+    virtual void SetTitle(const wxString& title) = 0;
+    virtual void SetIcon(const wxString& nsImageName) = 0;
     virtual bool SetIcon(const wxBitmapBundle& icon, const wxString& tooltip = wxEmptyString) = 0;
     virtual bool RemoveIcon() = 0;

@@ -93,6 +95,8 @@ class wxTaskBarIconDockImpl: public wxTaskBarIconImpl
 public:
     wxTaskBarIconDockImpl(wxTaskBarIcon *taskBarIcon);
     virtual ~wxTaskBarIconDockImpl();
+    virtual void SetTitle(const wxString& title) override;
+    virtual void SetIcon(const wxString& nsImageName) override;
     virtual bool SetIcon(const wxBitmapBundle& icon, const wxString& tooltip = wxEmptyString) override;
     virtual bool RemoveIcon() override;
     virtual bool PopupMenu(wxMenu *menu) override;
@@ -128,6 +132,8 @@ public:

     virtual bool IsStatusItem() const override { return true; }

+    virtual void SetTitle(const wxString& title) override;
+    virtual void SetIcon(const wxString& nsImageName) override;
     virtual bool SetIcon(const wxBitmapBundle& icon, const wxString& tooltip = wxEmptyString) override;
     virtual bool RemoveIcon() override;
     virtual bool PopupMenu(wxMenu *menu) override;
@@ -184,6 +190,18 @@ wxTaskBarIcon::~wxTaskBarIcon()
     }
 }
 
+void wxTaskBarIcon::SetTitle(const wxString& title)
+{
+    if ( m_impl )
+        m_impl->SetTitle(title);
+}
+
+void wxTaskBarIcon::SetIcon(const wxString& nsImageName)
+{
+    if ( m_impl )
+        m_impl->SetIcon(nsImageName);
+}
+
 bool wxTaskBarIcon::OSXIsStatusItem()
 {
     if ( m_impl )
@@ -290,6 +308,14 @@ WX_NSMenu wxTaskBarIconDockImpl::OSXDoGetDockHMenu()
     return (WX_NSMenu)m_pMenu->GetHMenu();
 }
 
+void wxTaskBarIconDockImpl::SetTitle(const wxString& WXUNUSED(title))
+{
+}
+
+void wxTaskBarIconDockImpl::SetIcon(const wxString& WXUNUSED(nsImageName))
+{
+}
+
 bool wxTaskBarIconDockImpl::SetIcon(const wxBitmapBundle& icon, const wxString& WXUNUSED(tooltip))
 {
     m_icon = IconFromBundle(icon);
@@ -372,6 +398,34 @@ wxTaskBarIconCustomStatusItemImpl::~wxTaskBarIconCustomStatusItemImpl()
 {
 }
 
+void wxTaskBarIconCustomStatusItemImpl::SetTitle(const wxString& title)
+{
+    wxCFStringRef cfTitle(title);
+    [[m_statusItem button] setTitle:cfTitle.AsNSString()];
+}
+
+void wxTaskBarIconCustomStatusItemImpl::SetIcon(const wxString& nsImageName)
+{
+    if(!m_statusItem)
+    {
+        m_statusItem = [[NSStatusBar systemStatusBar] statusItemWithLength:NSVariableStatusItemLength];
+        [m_statusItem retain];
+
+        m_target = [[wxOSXStatusItemTarget alloc] init];
+        [m_target setImplementation:this];
+        [[m_statusItem button] setTarget:m_target];
+        [[m_statusItem button] setAction:@selector(clickedAction:)];
+#if __MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_12
+        [[m_statusItem button] sendActionOn: NSEventMaskLeftMouseDown | NSEventMaskRightMouseDown];
+#endif
+    }
+
+    wxCFStringRef cfImageName(nsImageName);
+    NSImage* nsimage = [NSImage imageNamed: cfImageName.AsNSString()];
+    [nsimage setTemplate: YES];
+    [[m_statusItem button] setImage:nsimage];
+}
+
 bool wxTaskBarIconCustomStatusItemImpl::SetIcon(const wxBitmapBundle& icon, const wxString& tooltip)
 {
     if(!m_statusItem)
